lib/workers/repository/process/index.ts

```diff
@@ -2,23 +2,61 @@ import { mergeChildConfig } from '../../../config';
 import type { RenovateConfig } from '../../../config/types';
 import { logger } from '../../../logger';
 import type { PackageFile } from '../../../manager/types';
+import { platform } from '../../../platform';
+import { getCache } from '../../../util/cache/repository';
 import { branchExists } from '../../../util/git';
 import { addSplit } from '../../../util/split';
 import type { BranchConfig } from '../../types';
 import { readDashboardBody } from '../dependency-dashboard';
 import { ExtractResult, extract, lookup, update } from './extract-update';
 import type { WriteUpdateResult } from './write';
 
-function getBaseBranchConfig(
+async function getBaseBranchConfig(
   baseBranch: string,
   config: RenovateConfig
-): RenovateConfig {
+): Promise<RenovateConfig | null> {
   logger.debug(`baseBranch: ${baseBranch}`);
-  const baseBranchConfig = mergeChildConfig(config, { baseBranch });
-  if (config.baseBranches.length > 1) {
+
+  let baseBranchConfig: RenovateConfig = config;
+
+  if (
+    config.useBaseBranchConfig === 'replace' &&
+    baseBranch !== config.defaultBranch
+  ) {
+    logger.debug(
+      `Using config from branch ${baseBranch} because useBaseBranchConfig option specified`
+    );
+
+    // Retrieve config file name autodetected for this repo
+    const cache = getCache();
+    const configFileName =
+      cache.configFileName || config.onboardingConfigFileName;
+
+    try {
+      baseBranchConfig = await platform.getJsonFile(
+        configFileName,
+        config.repository,
+        baseBranch
+      );
+    } catch (err) {
+      logger.error(
+        `Error fetching config file ${configFileName} from branch ${baseBranch} - possible config name mismatch between branches?`
+      );
+      throw new Error(
+        `config error: Error fetching config file ${configFileName} from branch ${baseBranch}`
+      );
```

Comment:
Please throw like this:
https://github.com/renovatebot/renovate/blob/91842073ef58bf51ff9f492c8fce24c81f09fb23/lib/config/secrets.ts#L76-L82